From: Andrew Morton <akpm@linux-foundation.org>
Subject: sparc: add _mcount() prototype
Date: Fri May 22 09:20:26 PM PDT 2026

sparc64 defconfig told me

WARNING: modpost: EXPORT symbol "_mcount" [vmlinux] version generation failed, symbol will not be versioned.
Is "_mcount" prototyped in <asm/asm-prototypes.h>?

so I added it.

BTW, altering arch/sparc/include/asm/asm-prototypes.h then running `make'
doesn't compile anything, so there's a missing dependency somewhere?

Cc: "David S. Miller" <davem@davemloft.net>
Cc: Andreas Larsson <andreas@gaisler.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 arch/sparc/include/asm/asm-prototypes.h |    1 +
 1 file changed, 1 insertion(+)

--- a/arch/sparc/include/asm/asm-prototypes.h~sparc-add-_mcount-prototype
+++ a/arch/sparc/include/asm/asm-prototypes.h
@@ -25,6 +25,7 @@ void *memcpy(void *dest, const void *src
 void *memset(void *s, int c, size_t n);
 typedef int TItype __attribute__((mode(TI)));
 TItype __multi3(TItype a, TItype b);
+void _mcount(void);
 
 s64 __ashldi3(s64, int);
 s64 __lshrdi3(s64, int);
_
